src/components/protocol_handler/src/protocol_handler_impl.cc

```diff
@@ -839,6 +840,18 @@ void ProtocolHandlerImpl::OnConnectionClosed(
   multiframe_builder_.RemoveConnection(connection_id);
 }
 
+void ProtocolHandlerImpl::NotifyOnFailedHandshake() {
+  LOG4CXX_AUTO_TRACE(logger_);
+  sync_primitives::AutoLock lock(handshake_handlers_lock_);
+  std::list<std::shared_ptr<HandshakeHandler> >::iterator it =
+      handshake_handlers_.begin();
+  while (it != handshake_handlers_.end()) {
```

Comment:
@AByzhynar I woudl use for each loop, and swap with empty list after it. And why do we need to remove all hanshake handlers after failed handshake?

Reply:
@LuxoftAKutsan Can not do this as HandshakeHandler has no default ctor